internal/stores/settingsdb.go

```diff
@@ -0,0 +1,55 @@
+package stores
+
+import (
+	"errors"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// ErrSettingNotFound is returned if a specific setting can't be retrieved from the db.
+var ErrSettingNotFound = errors.New("setting not found")
+
+type (
+	dbSetting struct {
+		Model
+
+		Key   string `gorm:"unique;index"`
```

Comment:
I'd make this not null as well. To avoid accidentally setting an empty string as the key.